File: src/com/mishiranu/dashchan/util/FilenameUtils.java
package com.mishiranu.dashchan.util;

public class FilenameUtils {

    private static final String FILENAME_BLOCKED_CHARACTERS = "\\/:*?\"<>|.";
    private static final int FILENAME_MAX_CHARACTER_COUNT = 255;

    public static boolean isValidCharacter(char ch) {
        return !FILENAME_BLOCKED_CHARACTERS.contains(Character.toString(ch));
    }

    public static String getFilenameBlockedCharacters() {
        return FILENAME_BLOCKED_CHARACTERS;
    }

    public static int getFilenameMaxCharacterCount() {
        return FILENAME_MAX_CHARACTER_COUNT;
    }

}

File: src/com/mishiranu/dashchan/util/FilenameUtils.kt
package com.mishiranu.dashchan.util

object FilenameUtils {
    private const val FILENAME_BLOCKED_CHARACTERS = "\\/:*?\"<>|."
    private const val FILENAME_MAX_CHARACTER_COUNT = 255

    @JvmStatic
    fun isValidCharacter(ch: Char): Boolean = !FILENAME_BLOCKED_CHARACTERS.contains(ch)

    @JvmStatic
    fun getFilenameBlockedCharacters(): String = FILENAME_BLOCKED_CHARACTERS

    @JvmStatic
    fun getFilenameMaxCharacterCount(): Int = FILENAME_MAX_CHARACTER_COUNT
}
